lib/src/http/common/form_request.dart

```diff
@@ -39,19 +39,26 @@ abstract class CommonFormRequest extends CommonRequest implements FormRequest {
   MediaType get defaultContentType => new MediaType(
       'application', 'x-www-form-urlencoded', {'charset': encoding.name});
 
-  Map<String, String> get fields =>
+  Map<String, dynamic> get fields =>
       isSent ? new Map.unmodifiable(_fields) : _fields;
 
-  set fields(Map<String, String> fields) {
+  set fields(Map<String, dynamic> fields) {
     verifyUnsent();
     if (fields == null) {
       fields = {};
     }
     _fields = fields;
   }
 
-  Uint8List get _encodedQuery =>
-      encoding.encode(http_utils.mapToQuery(fields, encoding: encoding));
+  Uint8List get _encodedQuery {
+    fields.forEach((key, value) {
+      if (value is! String && value is! Iterable<String>) {
+        throw new ArgumentError('FormRequest: value of "$key" field must be of '
```

Comment:
Ideally, yes this check would be done immediately when adding or modifying fields, but in this case we're exposing a mutable map to allow for incremental additions like this:

```dart
var request = new FormRequest()
  ..fields['foo'] = 'bar'
  ..fields['bar'] = 'baz';
```
This doesn't actually use the setter - that would only happen if the consumer were to set the entire `fields` map. There's no way to catch additions like this. So I placed the check right before we encode the map to a String.